test_pull_requests

```diff
@@ -854,14 +898,22 @@ popd
       [updated_at, changed_after_eval]
     end
 
+    # add_coreq adds a co-requisite pull request to the repo_to_pull_request mapping if the co-requisite statement
+    # is trusted and the co-requisite pull request is merge-able.
     def add_coreq(addtl_pull_id, addtl_pull_repo, login, trigger_login, repo_to_pull_request, settings, trigger_updated_at, updated_at, base_pull_id, base_repo, comments)
       $stderr.puts "  Processing dependency on pull request #{addtl_pull_id} in repo '#{addtl_pull_repo}'"
-      trusted = user_trusted?(login, addtl_pull_repo, settings)
-      # It's ok if the user is trusted or the trigger was added after the comment
-      if trusted || (user_trusted?(trigger_login, addtl_pull_repo, settings) && (trigger_updated_at && trigger_updated_at >= updated_at))
+      # we can add the co-requisite if the author of the original pull request is trusted, or if the author of the trigger
+      # statement is trusted and they authored the trigger statement after the co-requisite statement was made
+      author_trusted = user_trusted?(login, addtl_pull_repo, settings)
+      trigger_author_trusted = user_trusted?(trigger_login, addtl_pull_repo, settings)
+      if author_trusted || (trigger_author_trusted && (trigger_updated_at && trigger_updated_at >= updated_at))
         pull_request = get_pull_request(addtl_pull_id, addtl_pull_repo, 2)
         if pull_request
-          if pull_request['mergeable']
+          # on co-requisite pull requests, we do not expect there to be trigger statements, as the 
+          # triggers on the parent pull request will trigger builds and tests with the co-requisites
+          # therefore, we need to manage the mergeable state of the co-requisite pull requests here
+          # while we're considering them for their parent
+          if is_mergeable?(base_pull_id, base_repo)
```

Comment:
This looks like you are checking against the wrong pull.  I am not sure if this is correct yet.  But can you move it to a separate pull to debate.